### PR TITLE
fix(ImmutableDevices): column spacing is fixed by extending column props

### PR DESCRIPTION
**Pull request by mkholjuraev**
# Description

Associated Jira ticket: # (issue)

Fixes spacing between groups and tags columns by extending the column prop with default prop in mergeAppColumns function.

# How to test the PR

Please include steps to test your PR.

# Before the change

image

# After the change

image

# Dependent work link

# Checklist:

- [x] The commit message has the Jira ticket linked
- [x] PR has a short description
- [x] Screenshots before and after the change are added
- [x] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

File: src/SmartComponents/HybridInventoryTabs/helpers.js [hunk 1]
```diff
@@ -87,3 +91,42 @@ export const useActionResolver = (handleModalToggle) =>
     ],
     []
   );
+
+export const useOnLoad = (filters) => {
+  const store = useStore();
+  return useCallback(
+    ({ mergeWithEntities, INVENTORY_ACTION_TYPES, mergeWithDetail }) => {
+      store.replaceReducer(
+        updateReducers({
```
Comment (bastilian) [inline src/SmartComponents/HybridInventoryTabs/helpers.js:100]: ^ Can this be tested as it is pointed out here?

Comment (mkholjuraev) [reply src/SmartComponents/HybridInventoryTabs/helpers.js:100]: @bastilian thank you for the review. I thought about this, but could not find a meaningful way of covering with tests. Happy to implement some suggestions.

Comment (bastilian) [reply src/SmartComponents/HybridInventoryTabs/helpers.js:100]: @mkholjuraev You could make the AsyncComponent mock in the ImmutableDevices test call the function provided via the `onLoad`. Than this code would be at least called. You could also add a separate test for the hook and assert that it does correctly "merge" and provides proper arguments to the `mergeWithEntities` function.

File: src/SmartComponents/HybridInventoryTabs/helpers.test.js [hunk 2]
```diff
@@ -144,3 +144,34 @@ describe('useActionResolver', () => {
...
+    );
+    expect(system_profile.props).toEqual({ width: 15, isStatic: true });
+  });
+  test('Should extend groups column props to disable sorting', () => {
+    const result = mergeAppColumns(defaultColumns);
+
+    const groups = result.find((column) => column.key === 'groups');
+    expect(groups.props).toEqual({ width: 15, isStatic: true });
+  });
+});
```
Comment (gkarat) [inline src/SmartComponents/HybridInventoryTabs/helpers.test.js:177]: Would be great if we could split the hook and tests related work into a separate commit. It confused me a bit why it's here whereas the PR only states that you fix the columns width

File: src/SmartComponents/HybridInventoryTabs/helpers.js [hunk 3]
```diff
@@ -87,3 +91,42 @@ export const useActionResolver = (handleModalToggle) =>
...
+};
+
+export const mergeAppColumns = (defaultColumns) => {
+  const lastSeenColumn = defaultColumns.find(({ key }) => key === 'updated');
+  const impacted_date = {
+    key: 'impacted_date',
+    title: 'First Impacted',
+    sortKey: 'impacted_date',
+    transforms: [sortable, wrappable],
+    props: { width: 15 },
```
Comment (gkarat) [inline src/SmartComponents/HybridInventoryTabs/helpers.js:120]: I am not sure this is the standard way we use to change the columns width. Shouldn't `cellWidth` transform property serve you better? https://www.patternfly.org/2022.11/components/table/react-legacy#controlling-text

File: src/SmartComponents/HybridInventoryTabs/helpers.test.js (hunk 2 above)
Comment (mkholjuraev) [reply src/SmartComponents/HybridInventoryTabs/helpers.test.js:177]: Hmm, I am not sure if this gives us any benefit? The PR is not large and furthermore once approved I will squash the PR into the master.

File: src/SmartComponents/HybridInventoryTabs/helpers.js (hunk 3 above)
Comment (mkholjuraev) [reply src/SmartComponents/HybridInventoryTabs/helpers.js:120]: As you can see these columns are passed to the inventory table in the middle. InventoTable then it gets mapped to cellWith down the line here: https://github.com/RedHatInsights/insights-inventory-frontend/blob/e9af8e577b5c088d4f377b11218393e7918ca9af/src/components/InventoryTable/helpers.js#L120

**Comment by mkholjuraev** [comment]
> How the Group and Tags columns gap was increased if you didn't modify their props in this PR anyhow? 🤔

They were affected because immutable uses groups column definitions coming as a part of defaultColumns and I was wrongly mutated it completely to make it static here: https://github.com/RedHatInsights/insights-advisor-frontend/pull/1201/files#diff-1c08cf0a43c16938d52ff334bd629f63d3efb4de80c72f9480b903f787893283L111. This PR fixes it by extending incoming column props with `isStatic: true`

File: src/SmartComponents/HybridInventoryTabs/helpers.test.js (hunk 2 above)
Comment (gkarat) [reply src/SmartComponents/HybridInventoryTabs/helpers.test.js:177]: OK, it was a nit-pick rather, feel free to move on

**Review by gkarat — DISMISSED** [review]
LGTM. this works as expected. I think it would be nice to improve tests as @bastilian suggested: https://github.com/RedHatInsights/insights-advisor-frontend/pull/1201#discussion_r1407622531. But giving this approval

**Comment by mkholjuraev** [comment]
@gkarat thank you for the review. I will add the suggested tests and then merge this.

**Comment by LiorKGOW** [comment]
Hey @mkholjuraev could you add the Jira ticket number please ?

**Comment by LiorKGOW** [comment]
Is this commit ready for a release @mkholjuraev ?

**Pull request merged**